### PR TITLE
Refactor navigation to community/user pages, and fixes to iOS swipe back not working properly

### PR DESCRIPTION
**Pull request by hjiangsu**
## Pull Request Description

- Refactors some code which handles navigation to CommunityPage/UserPage
  - `navigateToCommunityPage` and `navigateToUserPage` replaces the previous functions for navigating to community/user pages by name. It is now more generic and will accept the name or the id of the community/user
- Increased width for swipeable page route detection
- Moved swipeable page route offset back to default for iOS, but kept it the same for Android
- CreateCommentPage and CreatePostPage are no longer full screen swipe to go back. This fixes an issue where it is really difficult to horizontally scroll on the markdown options at the bottom of the screen 

Test out the following if possible!
- Navigating to community/user pages from all the possible places (from long press actions, tapping on usernames/communities, user and community sidebars, etc)
- Test out comment swipe detection to see if that's been improved (hopefully should trigger accidental comment swipes less often)

## Issue Being Fixed

Issue Number: N/A

## Checklist

- [ ] Did you update CHANGELOG.md?
- [ ] Did you use localized strings where applicable?
- [ ] Did you add `semanticLabel`s where applicable for accessibility?

### CODE REVIEW

**Comment by micahmo** [comment]
Let us know if this is ready for review!

**Comment by hjiangsu** [comment]
@micahmo Feel free to test this out! I think its pretty much good

File: lib/community/utils/post_card_action_helpers.dart [hunk 1]
```diff
@@ -183,41 +169,9 @@ void showPostActionBottomModalSheet(BuildContext context, PostViewMedia postView
...
-        providers: [
-          BlocProvider.value(value: accountBloc),
-          BlocProvider.value(value: authBloc),
-          BlocProvider.value(value: thunderBloc),
-        ],
-        child: CommunityPage(communityId: communityId),
-      ),
-    ),
-  );
+  navigateToCommunityPage(context, communityId: communityId);
```
Comment (micahmo) [inline lib/community/utils/post_card_action_helpers.dart:172]: Do we even need these wrapper functions any more, or could we just put `navigateToCommunityPage` right in the tap handler?

**Review by micahmo — APPROVED** [review]
The Code LGTM. I tested a few scenarios and they all worked fine!

**Comment by hjiangsu** [comment]
Alright, I'll go ahead and merge it in! Let me know if you encounter anything though 😅

**Pull request merged**